apps/readest-app/src/app/library/hooks/useBooksSync.ts

```diff
@@ -82,27 +86,46 @@ export const useBooksSync = ({ onSyncStart, onSyncEnd }: UseBooksSyncProps) => {
       return oldBook;
     };
 
-    const updatedLibrary = await Promise.all(library.map(processOldBook));
+    const oldBooksResults = await Promise.allSettled(library.map(processOldBook));
+    const updatedLibrary = oldBooksResults.map((result, index) => {
+      if (result.status === 'fulfilled') {
+        return result.value;
+      } else {
+        console.error('Failed to process old book:', library[index].title || library[index].hash, result.reason);
+        return library[index];
+      }
+    });
+
     const processNewBook = async (newBook: Book) => {
       if (!updatedLibrary.some((oldBook) => oldBook.hash === newBook.hash)) {
         if (newBook.uploadedAt && !newBook.deletedAt) {
           try {
             await appService?.downloadBook(newBook, true);
-          } catch {
-            console.error('Failed to download book:', newBook);
+          } catch (error) {
+            console.error('Failed to download book:', newBook.title || newBook.hash, error);
           } finally {
-            newBook.coverImageUrl = await appService?.generateCoverImageUrl(newBook);
```

Comment:
`generateCoverImageUrl` is only string manipulation and should never throw exception.